tests/publiccloud/cloud_netconfig.pm

```diff
@@ -12,77 +12,119 @@
 # Maintainer: qa-c team <qa-c@suse.de>
 
 use Mojo::Base 'publiccloud::basetest';
-use Test::Assert 'assert_equals';
+use Test::Assert qw(assert_equals assert_not_equals);
 use testapi;
 use publiccloud::utils qw(is_azure is_ec2 is_gce);
 
 sub run {
     my ($self, $args) = @_;
     my $instance = $args->{my_instance};
-    my $os_version = get_var('VERSION');
+    my $provider = $args->{my_provider};
+    $self->{instance} = $args->{my_instance};
     my $pers_net_rules = '/etc/udev/rules.d/75-persistent-net-generator.rules';
 
-    # Cloud metadata service API is reachable at local destination
-    # 169.254.169.254 in case of all public cloud providers.
-    my $pc_meta_api_ip = '169.254.169.254';
+    $instance->ssh_assert_script_run('systemctl is-enabled cloud-netconfig.service');
 
-    my $cloud_netconfig_svc_status = $instance->ssh_script_run(
-        cmd => 'systemctl status cloud-netconfig.service');
-    if ($cloud_netconfig_svc_status == 0 || $cloud_netconfig_svc_status == 3)
-    {
-        record_info('Success', 'Status of cloud-netconfig.service is OK.');
-    }
-    else {
-        die('cloud-netconfig.service status appears to be incorrect!');
-    }
-
-    $instance->ssh_assert_script_run(
-        cmd => 'systemctl status cloud-netconfig.timer',
-        fail_message => 'cloud-netconfig.timer appears not to be started.'
-    );
+    $instance->ssh_assert_script_run('systemctl status cloud-netconfig.timer');
 
     # 75-persistent-net-generator.rules is usually a symlink to /dev/null in SLES 12+
-    $instance->ssh_assert_script_run(
-        cmd => "test -L $pers_net_rules",
-        fail_message => "File \"$pers_net_rules\" should be a symlink to "
-          . "\"/dev/null\" in SLES 12+."
-    );
-
-    my $grep_cloudvm_ipv4_cmd =
-      'ip -4 addr show eth0 | grep -oP "^\s+inet\s(\d+\.){3}\d+" | '
-      . 'sed "s/ *inet //g"';
-    my $local_eth0_ip =
-      $instance->ssh_script_output(cmd => $grep_cloudvm_ipv4_cmd);
+    $instance->ssh_assert_script_run("test -L $pers_net_rules");
+
+    record_info('eth0', 'Get public IP address for eth0');
+    my $local_eth0_ip = $instance->ssh_script_output(qq(ip -4 -o a s eth0 primary | grep -Po "inet \\K[\\d.]+" | head -n1));
     chomp($local_eth0_ip);
+    my $metadata_eth0_ip = $provider->query_metadata($instance, ifNum => '0', addrCount => '0');
+    die("Failed to get interface 0 IP from metadata server") unless length($metadata_eth0_ip);
```

Comment:
```suggestion
    die("Failed to get interface IPs from metadata server") unless length($metadata_eth0_ip);
```
I would remove the interface numbers here, as they are confusing to anyone who has not the full code in their head. I think this makes a better error message, because it still explains what happens but without unnecessary (and confusing) interface indices.

Reply:
I applied your suggestion and moved the check inside `$provider->query_metadata()`.